### PR TITLE
Ofmcc 851 application table actions

### PR DESCRIPTION
**Pull request by vietle-cgi**
Add Cancel application functionality.
Move ApplicationView to applications folder.
Add routing for Application/Facility Details page.

### CODE REVIEW

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/router/index.js [hunk 1]
```diff
@@ -114,6 +116,15 @@ const router = createRouter({
         requiresAuth: true,
       },
     },
+    {
+      path: getApplicationUrl(PATHS.FACILITY_DETAILS, ':applicationGuid'),
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/router/index.js]: it is fixed

File: frontend/src/components/applications/ApplicationsView.vue [hunk 2]
```diff
@@ -5,7 +5,7 @@
       <div v-if="isEmpty(applications)">You have no applications on file</div>
       <v-data-table v-else :headers="headers" :items="applications" item-key="applicationId" density="compact">
         <template #item.actions="{ item }">
-          <router-link :to="{ name: 'home' }">
+          <router-link :to="getRoutingPath(item?.applicationId)">
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/components/applications/ApplicationsView.vue]: it is fixed

File: frontend/src/components/applications/ApplicationsView.vue [hunk 3]
```diff
@@ -99,4 +103,9 @@ export default {
 :deep(div.v-data-table-footer) {
   padding-top: 20px;
 }
+:deep(.v-data-table-header__content:hover) {
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/components/applications/ApplicationsView.vue]: it is fixed

File: frontend/src/components/applications/ApplicationsView.vue [hunk 4]
```diff
@@ -5,7 +5,7 @@
       <div v-if="isEmpty(applications)">You have no applications on file</div>
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/components/applications/ApplicationsView.vue]: it is fixed

File: frontend/src/components/applications/FacilityDetails.vue [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<template>
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/components/applications/FacilityDetails.vue]: it is fixed

File: frontend/src/stores/applications.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        const response = await ApiService.apiAxios.get(ApiRoutes.APPLICATIONS + '/facility/' + facilityId)
+        return response?.data
+      } catch (error) {
+        console.log(`Failed to get the list of applications by facility id - ${error}`)
+        throw error
+      }
+    },
+
+    async getApplications() {
+      try {
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/stores/applications.js]: it is fixed

File: frontend/src/components/applications/CancelApplicationDialog.vue [hunk 7]
```diff
@@ -0,0 +1,73 @@
+<template>
+  <AppDialog v-model="isDisplayed" title="Confirm" :isLoading="isLoading" persistent max-width="50%" @close="closeDialog">
```
Comment (weskubo-cgi) [inline frontend/src/components/applications/CancelApplicationDialog.vue:2]: Let's add the min-width="350px" to this.

File: frontend/src/components/applications/CancelApplicationDialog.vue [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    <template #content>
+      <div align="center" class="confirm-dialog-text">Are you sure you want to cancel this application?</div>
+    </template>
+    <template #button>
+      <v-row class="mt-2" justify="space-around">
+        <AppButton id="dialog-go-back" :primary="false" size="large" width="200px" :loading="isLoading" @click="closeDialog">Go back</AppButton>
+        <AppButton id="dialog-cancel-application" size="large" width="200px" :loading="isLoading" @click="cancel">Cancel application</AppButton>
```
Comment (weskubo-cgi) [inline frontend/src/components/applications/CancelApplicationDialog.vue:9]: Button size is a little small.

File: frontend/src/components/applications/CancelApplicationDialog.vue [hunk 9]
```diff
@@ -0,0 +1,73 @@
+<template>
+  <AppDialog v-model="isDisplayed" title="Confirm" :isLoading="isLoading" persistent max-width="50%" @close="closeDialog">
+    <template #content>
+      <div align="center" class="confirm-dialog-text">Are you sure you want to cancel this application?</div>
+    </template>
+    <template #button>
+      <v-row class="mt-2" justify="space-around">
```
Comment (weskubo-cgi) [inline frontend/src/components/applications/CancelApplicationDialog.vue:7]: We should add v-cols to align better at low resolution. See what was done here https://github.com/bcgov/ecc-ofm/blob/ofmcc-729-update-user/frontend/src/components/settings/ManageUserDialog.vue.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/applications/CancelApplicationDialog.vue (hunk 7 above)
Comment (vietle-cgi) [reply frontend/src/components/applications/CancelApplicationDialog.vue:2]: I added min-width to AppDialog component

File: frontend/src/components/applications/CancelApplicationDialog.vue (hunk 8 above)
Comment (vietle-cgi) [reply frontend/src/components/applications/CancelApplicationDialog.vue]: it is fixed

File: frontend/src/components/applications/CancelApplicationDialog.vue (hunk 9 above)
Comment (vietle-cgi) [reply frontend/src/components/applications/CancelApplicationDialog.vue]: it is fixed

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**